src/pages/cli/graphql/examples-and-solutions.mdx

```diff
@@ -564,4 +564,83 @@ query repsByPeriodAndTotal {
     }
   }
 }
-```
\ No newline at end of file
+```
+
+## Advanced
+
+### Batch Put Custom Resolver
+
+Sometimes it's preferred to create objects in bulk, rather than subsequently creating individual objects and waiting for all the requests to complete.
+
+1. Define your schema with a custom mutation (the custom mutation should not be deployed to AppSync beforehand if following these steps)
+```graphql
+type Todo @model @auth(rules: [{allow: private, provider: iam}]) {
+  id: ID!
+  name: String!
+  description: String
+}
+
+type Mutation {
+  batchCreateTodo(todos: [BatchCreateTodo]): [Todo]
+}
+
+input BatchCreateTodo {
+  id: ID
+  name: String!
+  description: String
+}
+```
+
+2. Before pushing the new custom mutation, [create a custom resource for your resolver](/cli/graphql/custom-business-logic/#vtl-resolver) and use the following code snippets as a guide to get started with your VTL
+
+3. Use the following imports as a starting point for your custom CDK stack
+```typescript
+import * as cdk from '@aws-cdk/core';
+import * as AmplifyHelpers from '@aws-amplify/cli-extensibility-helper';
+import * as appsync from '@aws-cdk/aws-appsync';
+import { AmplifyDependentResourcesAttributes } from '../../types/amplify-dependent-resources-ref';
+import { stateManager } from 'amplify-cli-core';
```

Comment:
I don't think you're supposed to import the state manager, right? Shouldn't you use the Amplify Helpers function?